plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemorySplitManager.java

```diff
@@ -72,6 +77,45 @@ public ConnectorSplitSource getSplits(
                 splits.add(new MemorySplit(table.getId(), i, splitsPerNode, dataFragment.getHostAddress(), rows, OptionalLong.empty()));
             }
         }
-        return new FixedSplitSource(splits.build());
+
+        ConnectorSplitSource splitSource = new FixedSplitSource(splits.build());
+        if (enableLazyDynamicFiltering) {
+            splitSource = new DelayedSplitSource(dynamicFilter.isBlocked(), splitSource);
+        }
+        return splitSource;
+    }
+
+    private static class DelayedSplitSource
```

Comment:
It would be great to add a code comment here why the delay is needed. Unlike other connectors we don't actually use the DF here for pruning splits. Since MemoryPageSourceProvider will also continue to do the same waiting on DFs (because we need to wait for DFs to be delivered to the workers for deterministic results), it will be hard to understand why we have this delay without prior context.

Reply:
Added a comment